quickjs.c

```diff
@@ -20972,6 +20972,14 @@ static __exception int js_parse_class(JSParseState *s, BOOL is_class_expr,
             emit_atom(s, JS_ATOM_this);
             emit_u16(s, 0);
 
+            // expose class name to static initializers
+            if (is_static) {
+                emit_op(s, OP_dup);
+                emit_op(s, OP_scope_put_var_init);
+                emit_atom(s, class_name1);
+                emit_u16(s, s->cur_func->scope_level);
+            }
```

Comment:
On second thought...
```suggestion
            if (is_static && class_name != JS_ATOM_NULL) {
                emit_op(s, OP_dup);
                emit_op(s, OP_scope_put_var_init);
                emit_atom(s, class_name);
                emit_u16(s, s->cur_func->scope_level);
            }
```
...because `class_name1` can be computed or the empty string.